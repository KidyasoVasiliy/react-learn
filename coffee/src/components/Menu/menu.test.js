import React from 'react'
import { Menu } from './index';
import { shallow } from 'enzyme';

describe('Testing <Menu />', () => {
  it('Menu have renderer correctly', () => {
    const menu = shallow(<Menu />);
    expect(menu).toMatchSnapshot();
  })
});
