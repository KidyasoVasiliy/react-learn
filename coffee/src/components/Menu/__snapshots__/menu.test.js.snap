// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Menu /> Menu have renderer correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Menu />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          className="nav__link"
          replace={false}
          to="/"
        >
          Coffee house
        </Link>,
        <Link
          className="nav__link"
          replace={false}
          to="/our-coffee"
        >
          Our coffee
        </Link>,
        <Link
          className="nav__link"
          replace={false}
          to="/for-your-pleasure"
        >
          For your pleasure
        </Link>,
      ],
      "className": "nav undefined",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Coffee house",
          "className": "nav__link",
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": "Coffee house",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Our coffee",
          "className": "nav__link",
          "replace": false,
          "to": "/our-coffee",
        },
        "ref": null,
        "rendered": "Our coffee",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "For your pleasure",
          "className": "nav__link",
          "replace": false,
          "to": "/for-your-pleasure",
        },
        "ref": null,
        "rendered": "For your pleasure",
        "type": [Function],
      },
    ],
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            className="nav__link"
            replace={false}
            to="/"
          >
            Coffee house
          </Link>,
          <Link
            className="nav__link"
            replace={false}
            to="/our-coffee"
          >
            Our coffee
          </Link>,
          <Link
            className="nav__link"
            replace={false}
            to="/for-your-pleasure"
          >
            For your pleasure
          </Link>,
        ],
        "className": "nav undefined",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Coffee house",
            "className": "nav__link",
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": "Coffee house",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Our coffee",
            "className": "nav__link",
            "replace": false,
            "to": "/our-coffee",
          },
          "ref": null,
          "rendered": "Our coffee",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "For your pleasure",
            "className": "nav__link",
            "replace": false,
            "to": "/for-your-pleasure",
          },
          "ref": null,
          "rendered": "For your pleasure",
          "type": [Function],
        },
      ],
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
